### DISCUSSION TITLE
How can I create a legacy URL for a section whose name has changed?

### DISCUSSION THREAD

**Discussion by mhucka** [executablebooks/MyST-Parser#767] [Q&A] [answered]
I wrote some software docs using MyST. I updated the docs and changed some section headings. Some of the URLs pointing to sections, i.e., URLs of the form
```
https://foo.github.io/software/page.html#old-section-name
```
no longer work because when I changed the name of the sections, the auto-generated `id` values changed too.

Is there a way to retain (for lack of a better term) legacy id's on the sections, or somehow remap old URLs to new URLs, for cases like this?

**Comment by chrisjsewell** [comment, marked as answer]
I'm afraid not at present, this is why it is recommended to use explicit targets which can remain the same across section changes: https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#creating-explicit-targets

**Comment by mhucka** [comment]
I ended up doing that, and it works. Thanks for the pointer. In lieu of an automated method, this works well enough (for my needs, anyway).
